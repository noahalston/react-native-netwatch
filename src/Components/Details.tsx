import * as React from 'react';
import { StyleSheet, View, ScrollView } from 'react-native';
import { Appbar, Subheading, Text, Surface } from 'react-native-paper';
import { Status } from './Status';
import { duration, convert, getDate } from '../Utils/helpers';
import { Request } from '../Core/Request';
// @ts-ignore
import BlobFileReader from 'react-native/Libraries/Blob/FileReader';

interface IProps {
  testId?: string;
  onPressBack: (showDetails: boolean) => void;
  item?: Request;
}

// These attribute will be not added in the detail's scrollview because always displayed in the other components
const excludedAttributes: Array<string> = [
  '_id',
  'readyState',
  'method',
  'status',
  'startTime',
  'endTime',
  'dataSent',
  'requestHeaders',
  'responseHeaders',
  'response',
  'responseSize',
  'responseContentType',
];

const _renderItems = (listOfItems: Array<[string, any]>) => {
  return listOfItems
    .filter((item: Array<string>) => !excludedAttributes.includes(item[0]))
    .map((item: Array<string>, index: number) => {
      return (
        <View key={index} style={styles.attribtuesContainer}>
          <Text style={styles.attributes}>{item[0]}</Text>
          <Text style={styles.text}>{item[1]}</Text>
        </View>
      );
    });
};

export const Details: React.FC<IProps> = (props) => {
  if (!props.item) return <Text>Error</Text>;
  // Appbar header is repeated here cause we use the absolute position in the style
  // Put this directly in the index.tsx cause that the Appbar will be added
  return (
    <View style={styles.container}>
      <Appbar.Header>
        <Appbar.BackAction
          onPress={() => props.onPressBack(false)}
          testID="buttonBackToMainScreen"
        />
        <Appbar.Content title="Netwatch" />
      </Appbar.Header>
      <Surface style={{ flexDirection: 'row' }}>
        {props.item && <Status item={props.item} />}
        <View style={{ justifyContent: 'center' }}>
          <Text style={styles.textSubheader}>{`Started at: ${getDate(props.item.startTime)}`}</Text>
          <Text style={styles.textSubheader}>{`Duration ${convert(
            duration(props.item.startTime, props.item.endTime)
          )}ms`}</Text>
        </View>
      </Surface>
      <View style={{ flex: 1 }}>
        <ScrollView contentContainerStyle={styles.scrollview}>
          <View>
            <Subheading style={styles.subheading}>GENERAL</Subheading>
            {props.item && _renderItems(Object.entries(props.item))}
            <Subheading style={styles.subheading}>REQUEST</Subheading>
            {props.item?.requestHeaders && _renderItems(Object.entries(props.item.requestHeaders))}
            <Subheading style={styles.subheading}>BODY REQUEST</Subheading>
            <View style={styles.attribtuesContainer}>
              <Text style={styles.text}>{props.item.dataSent}</Text>
            </View>
            <Subheading style={styles.subheading}>RESPONSE</Subheading>
            {props.item &&
              props.item?.responseHeaders &&
              _renderItems(Object.entries(props.item.responseHeaders))}
            <Subheading style={styles.subheading}>BODY RESPONSE</Subheading>
            <View style={styles.attribtuesContainer}>
              <Text style={styles.text}>{props.item.response}</Text>
            </View>
          </View>
        </ScrollView>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    position: 'absolute',
    height: '100%',
    width: '100%',
    flex: 1,
    top: 0,
    left: 0,
    backgroundColor: 'white',
  },
  subheading: {
    fontWeight: 'bold',
    paddingHorizontal: 16,
    paddingVertical: 8,
    borderWidth: 0,
    marginTop: -1,
    backgroundColor: 'lightgray',
    color: 'black',
  },
  attribtuesContainer: {
    paddingHorizontal: 16,
    paddingVertical: 8,
  },
  attributes: {
    fontSize: 16,
    fontWeight: 'bold',
    color: 'black',
  },
  text: {
    fontSize: 16,
    color: 'black',
  },
  textSubheader: {
    fontSize: 16,
  },
  scrollview: {
    paddingBottom: 20,
  },
});
