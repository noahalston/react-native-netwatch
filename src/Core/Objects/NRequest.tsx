import { ILog, LogType } from '../../types';

export class NRequest implements ILog {
  _id: number = -1;
  type: LogType = 'NR';
  startTime: number = Date.now();
  readyState: number = 0;
  url: string = '';
  shortUrl?: string = '';
  method: string = '';
  status: number = -1;
  endTime: number = 0;
  timeout?: number;
  dataSent: string = '';
  requestHeaders?: any;
  responseHeaders?: any;
  responseContentType?: string;
  responseSize?: number;
  responseType?: string;
  responseURL?: string;
  response: string = '';

  constructor(attributes?: any) {
    Object.assign(this, attributes);
  }
}

export default NRequest;
