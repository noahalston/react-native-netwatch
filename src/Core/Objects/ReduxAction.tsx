import { ILog, LogType } from '../../types';

interface IAction {
  type: string;
  payload: any;
}

export class ReduxAction implements ILog {
  _id: number = -1;
  startTime: number = Date.now();
  type: LogType = 'REDUX';
  action: IAction = { type: '__ERROR:UNDEFINED__', payload: '' };

  constructor(attributes?: any) {
    Object.assign(this, attributes);
  }
}

export default ReduxAction;
