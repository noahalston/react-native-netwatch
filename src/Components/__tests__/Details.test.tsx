import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { Details, IProps } from '../Details';
import { RNRequest } from '../../Core/Objects/RNRequest';

describe('Details test suite', () => {
  let component: ShallowWrapper;
  let props: IProps;
  const back = jest.fn();

  it('should render properly', () => {
    givenProps();
    givenComponent();
    expect(component).toMatchSnapshot();
  });

  it('should render properly with request without headers & press backbutton', () => {
    givenProps(mockRequest);
    givenComponent();
    expect(component).toMatchSnapshot();

    whenPressingButton('buttonBackToMainScreen');
    expect(back).toHaveBeenCalledTimes(1);
  });

  it('should render properly with request with headers', () => {
    givenProps(mockRequestWithHeaders);
    givenComponent();
    expect(component).toMatchSnapshot();
  });

  // GIVEN
  function givenComponent() {
    component = shallow(<Details {...props} />);
  }

  function givenProps(item?: RNRequest) {
    props = {
      testId: 'testId',
      onPressBack: back,
      item,
    };
  }

  // UTILITIES

  function whenPressingButton(testId: string) {
    component.find(`[testID="${testId}"]`).simulate('press');
  }

  const mockRequest: RNRequest = {
    _id: 73,
    dataSent: 'dataSent',
    endTime: 1613477575757,
    method: 'GET',
    readyState: 4,
    response: 'response',
    responseContentType: 'application/json',
    responseSize: 0,
    responseType: 'blob',
    responseURL: 'https://run.mocky.io/v3/1a2d092a-42b2-4a89-a44f-267935dc13e9',
    startTime: 1613477574742,
    status: 200,
    timeout: 0,
    type: 'RNR',
    url: 'https://run.mocky.io/v3/1a2d092a-42b2-4a89-a44f-267935dc13e9',
  };

  const mockRequestWithHeaders: RNRequest = {
    _id: 73,
    dataSent: 'dataSent',
    endTime: 1613477575757,
    method: 'GET',
    readyState: 4,
    requestHeaders: {
      'Content-Type': 'application/json',
    },
    response: 'response',
    responseContentType: 'application/json',
    responseHeaders: {
      'Content-Length': '0',
      'Content-Type': 'application/json; charset=UTF-8',
      Date: 'Tue, 16 Feb 2021 12:12:55 GMT',
      'Sozu-Id': '51989c0c-ebe7-4574-913d-443477875da7',
    },
    responseSize: 0,
    responseType: 'blob',
    responseURL: 'https://run.mocky.io/v3/1a2d092a-42b2-4a89-a44f-267935dc13e9',
    startTime: 1613477574742,
    status: 200,
    timeout: 0,
    type: 'RNR',
    url: 'https://run.mocky.io/v3/1a2d092a-42b2-4a89-a44f-267935dc13e9',
  };
});
