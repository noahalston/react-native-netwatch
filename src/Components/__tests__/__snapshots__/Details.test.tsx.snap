// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details test suite should render properly 1`] = `
<ForwardRef(withTheme(undefined))>
  Error
</ForwardRef(withTheme(undefined))>
`;

exports[`Details test suite should render properly with request with headers 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "height": "100%",
      "left": 0,
      "position": "absolute",
      "top": 0,
      "width": "100%",
    }
  }
>
  <ForwardRef(withTheme(Appbar.Header))
    style={
      Object {
        "backgroundColor": "#212121",
      }
    }
  >
    <Appbar.BackAction
      color="white"
      onPress={[Function]}
      testID="buttonBackToMainScreen"
    />
    <ForwardRef(withTheme(Appbar.Content))
      title="Netwatch"
    />
  </ForwardRef(withTheme(Appbar.Header))>
  <ForwardRef(withTheme(Surface))
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Status
      item={
        Object {
          "_id": 73,
          "dataSent": "dataSent",
          "endTime": 1613477575757,
          "method": "GET",
          "readyState": 4,
          "requestHeaders": Object {
            "Content-Type": "application/json",
          },
          "response": "response",
          "responseContentType": "application/json",
          "responseHeaders": Object {
            "Content-Length": "0",
            "Content-Type": "application/json; charset=UTF-8",
            "Date": "Tue, 16 Feb 2021 12:12:55 GMT",
            "Sozu-Id": "51989c0c-ebe7-4574-913d-443477875da7",
          },
          "responseSize": 0,
          "responseType": "blob",
          "responseURL": "https://run.mocky.io/v3/1a2d092a-42b2-4a89-a44f-267935dc13e9",
          "startTime": 1613477574742,
          "status": 200,
          "timeout": 0,
          "type": "RNR",
          "url": "https://run.mocky.io/v3/1a2d092a-42b2-4a89-a44f-267935dc13e9",
        }
      }
    />
    <Component
      style={
        Object {
          "backgroundColor": "#212121",
          "justifyContent": "center",
          "paddingLeft": 8,
          "width": "100%",
        }
      }
    >
      <ForwardRef(withTheme(undefined))
        style={
          Object {
            "color": "white",
            "fontSize": 16,
          }
        }
      >
        Tue Feb 16 2021 12:12:54 GMT+0000 (Coordinated Universal Time)
      </ForwardRef(withTheme(undefined))>
      <ForwardRef(withTheme(undefined))
        style={
          Object {
            "color": "white",
            "fontSize": 16,
          }
        }
      >
        Duration 1015ms
      </ForwardRef(withTheme(undefined))>
    </Component>
  </ForwardRef(withTheme(Surface))>
  <Component
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Component
      contentContainerStyle={
        Object {
          "paddingBottom": 20,
        }
      }
    >
      <Component>
        <Subheading
          style={
            Object {
              "backgroundColor": "lightgray",
              "borderWidth": 0,
              "color": "black",
              "fontWeight": "bold",
              "marginTop": -1,
              "paddingHorizontal": 16,
              "paddingVertical": 8,
            }
          }
        >
          GENERAL
        </Subheading>
        <Component
          key="0"
          style={
            Object {
              "paddingHorizontal": 16,
              "paddingVertical": 8,
            }
          }
        >
          <ForwardRef(withTheme(undefined))
            style={
              Object {
                "color": "black",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            responseURL
          </ForwardRef(withTheme(undefined))>
          <ForwardRef(withTheme(undefined))
            style={
              Object {
                "color": "black",
                "fontSize": 16,
              }
            }
          >
            https://run.mocky.io/v3/1a2d092a-42b2-4a89-a44f-267935dc13e9
          </ForwardRef(withTheme(undefined))>
        </Component>
        <Component
          key="1"
          style={
            Object {
              "paddingHorizontal": 16,
              "paddingVertical": 8,
            }
          }
        >
          <ForwardRef(withTheme(undefined))
            style={
              Object {
                "color": "black",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            timeout
          </ForwardRef(withTheme(undefined))>
          <ForwardRef(withTheme(undefined))
            style={
              Object {
                "color": "black",
                "fontSize": 16,
              }
            }
          >
            0
          </ForwardRef(withTheme(undefined))>
        </Component>
        <Component
          key="2"
          style={
            Object {
              "paddingHorizontal": 16,
              "paddingVertical": 8,
            }
          }
        >
          <ForwardRef(withTheme(undefined))
            style={
              Object {
                "color": "black",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            url
          </ForwardRef(withTheme(undefined))>
          <ForwardRef(withTheme(undefined))
            style={
              Object {
                "color": "black",
                "fontSize": 16,
              }
            }
          >
            https://run.mocky.io/v3/1a2d092a-42b2-4a89-a44f-267935dc13e9
          </ForwardRef(withTheme(undefined))>
        </Component>
        <Subheading
          style={
            Object {
              "backgroundColor": "lightgray",
              "borderWidth": 0,
              "color": "black",
              "fontWeight": "bold",
              "marginTop": -1,
              "paddingHorizontal": 16,
              "paddingVertical": 8,
            }
          }
        >
          REQUEST HEADERS
        </Subheading>
        <Component
          key="0"
          style={
            Object {
              "paddingHorizontal": 16,
              "paddingVertical": 8,
            }
          }
        >
          <ForwardRef(withTheme(undefined))
            style={
              Object {
                "color": "black",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Content-Type
          </ForwardRef(withTheme(undefined))>
          <ForwardRef(withTheme(undefined))
            style={
              Object {
                "color": "black",
                "fontSize": 16,
              }
            }
          >
            application/json
          </ForwardRef(withTheme(undefined))>
        </Component>
        <Subheading
          style={
            Object {
              "backgroundColor": "lightgray",
              "borderWidth": 0,
              "color": "black",
              "fontWeight": "bold",
              "marginTop": -1,
              "paddingHorizontal": 16,
              "paddingVertical": 8,
            }
          }
        >
          REQUEST DATA
        </Subheading>
        <Component
          style={
            Object {
              "paddingHorizontal": 16,
              "paddingVertical": 8,
            }
          }
        >
          <ForwardRef(withTheme(undefined))
            style={
              Object {
                "color": "black",
                "fontSize": 16,
              }
            }
          >
            dataSent
          </ForwardRef(withTheme(undefined))>
        </Component>
        <Subheading
          style={
            Object {
              "backgroundColor": "lightgray",
              "borderWidth": 0,
              "color": "black",
              "fontWeight": "bold",
              "marginTop": -1,
              "paddingHorizontal": 16,
              "paddingVertical": 8,
            }
          }
        >
          RESPONSE HEADERS
        </Subheading>
        <Component
          key="0"
          style={
            Object {
              "paddingHorizontal": 16,
              "paddingVertical": 8,
            }
          }
        >
          <ForwardRef(withTheme(undefined))
            style={
              Object {
                "color": "black",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Content-Length
          </ForwardRef(withTheme(undefined))>
          <ForwardRef(withTheme(undefined))
            style={
              Object {
                "color": "black",
                "fontSize": 16,
              }
            }
          >
            0
          </ForwardRef(withTheme(undefined))>
        </Component>
        <Component
          key="1"
          style={
            Object {
              "paddingHorizontal": 16,
              "paddingVertical": 8,
            }
          }
        >
          <ForwardRef(withTheme(undefined))
            style={
              Object {
                "color": "black",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Content-Type
          </ForwardRef(withTheme(undefined))>
          <ForwardRef(withTheme(undefined))
            style={
              Object {
                "color": "black",
                "fontSize": 16,
              }
            }
          >
            application/json; charset=UTF-8
          </ForwardRef(withTheme(undefined))>
        </Component>
        <Component
          key="2"
          style={
            Object {
              "paddingHorizontal": 16,
              "paddingVertical": 8,
            }
          }
        >
          <ForwardRef(withTheme(undefined))
            style={
              Object {
                "color": "black",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Date
          </ForwardRef(withTheme(undefined))>
          <ForwardRef(withTheme(undefined))
            style={
              Object {
                "color": "black",
                "fontSize": 16,
              }
            }
          >
            Tue, 16 Feb 2021 12:12:55 GMT
          </ForwardRef(withTheme(undefined))>
        </Component>
        <Component
          key="3"
          style={
            Object {
              "paddingHorizontal": 16,
              "paddingVertical": 8,
            }
          }
        >
          <ForwardRef(withTheme(undefined))
            style={
              Object {
                "color": "black",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Sozu-Id
          </ForwardRef(withTheme(undefined))>
          <ForwardRef(withTheme(undefined))
            style={
              Object {
                "color": "black",
                "fontSize": 16,
              }
            }
          >
            51989c0c-ebe7-4574-913d-443477875da7
          </ForwardRef(withTheme(undefined))>
        </Component>
        <Subheading
          style={
            Object {
              "backgroundColor": "lightgray",
              "borderWidth": 0,
              "color": "black",
              "fontWeight": "bold",
              "marginTop": -1,
              "paddingHorizontal": 16,
              "paddingVertical": 8,
            }
          }
        >
          RESPONSE BODY
        </Subheading>
        <Component
          style={
            Object {
              "paddingHorizontal": 16,
              "paddingVertical": 8,
            }
          }
        >
          <ForwardRef(withTheme(undefined))
            style={
              Object {
                "color": "black",
                "fontSize": 16,
              }
            }
          >
            response
          </ForwardRef(withTheme(undefined))>
        </Component>
      </Component>
    </Component>
  </Component>
</Component>
`;

exports[`Details test suite should render properly with request without headers & press backbutton 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "height": "100%",
      "left": 0,
      "position": "absolute",
      "top": 0,
      "width": "100%",
    }
  }
>
  <ForwardRef(withTheme(Appbar.Header))
    style={
      Object {
        "backgroundColor": "#212121",
      }
    }
  >
    <Appbar.BackAction
      color="white"
      onPress={[Function]}
      testID="buttonBackToMainScreen"
    />
    <ForwardRef(withTheme(Appbar.Content))
      title="Netwatch"
    />
  </ForwardRef(withTheme(Appbar.Header))>
  <ForwardRef(withTheme(Surface))
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Status
      item={
        Object {
          "_id": 73,
          "dataSent": "dataSent",
          "endTime": 1613477575757,
          "method": "GET",
          "readyState": 4,
          "response": "response",
          "responseContentType": "application/json",
          "responseSize": 0,
          "responseType": "blob",
          "responseURL": "https://run.mocky.io/v3/1a2d092a-42b2-4a89-a44f-267935dc13e9",
          "startTime": 1613477574742,
          "status": 200,
          "timeout": 0,
          "type": "RNR",
          "url": "https://run.mocky.io/v3/1a2d092a-42b2-4a89-a44f-267935dc13e9",
        }
      }
    />
    <Component
      style={
        Object {
          "backgroundColor": "#212121",
          "justifyContent": "center",
          "paddingLeft": 8,
          "width": "100%",
        }
      }
    >
      <ForwardRef(withTheme(undefined))
        style={
          Object {
            "color": "white",
            "fontSize": 16,
          }
        }
      >
        Tue Feb 16 2021 12:12:54 GMT+0000 (Coordinated Universal Time)
      </ForwardRef(withTheme(undefined))>
      <ForwardRef(withTheme(undefined))
        style={
          Object {
            "color": "white",
            "fontSize": 16,
          }
        }
      >
        Duration 1015ms
      </ForwardRef(withTheme(undefined))>
    </Component>
  </ForwardRef(withTheme(Surface))>
  <Component
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Component
      contentContainerStyle={
        Object {
          "paddingBottom": 20,
        }
      }
    >
      <Component>
        <Subheading
          style={
            Object {
              "backgroundColor": "lightgray",
              "borderWidth": 0,
              "color": "black",
              "fontWeight": "bold",
              "marginTop": -1,
              "paddingHorizontal": 16,
              "paddingVertical": 8,
            }
          }
        >
          GENERAL
        </Subheading>
        <Component
          key="0"
          style={
            Object {
              "paddingHorizontal": 16,
              "paddingVertical": 8,
            }
          }
        >
          <ForwardRef(withTheme(undefined))
            style={
              Object {
                "color": "black",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            responseURL
          </ForwardRef(withTheme(undefined))>
          <ForwardRef(withTheme(undefined))
            style={
              Object {
                "color": "black",
                "fontSize": 16,
              }
            }
          >
            https://run.mocky.io/v3/1a2d092a-42b2-4a89-a44f-267935dc13e9
          </ForwardRef(withTheme(undefined))>
        </Component>
        <Component
          key="1"
          style={
            Object {
              "paddingHorizontal": 16,
              "paddingVertical": 8,
            }
          }
        >
          <ForwardRef(withTheme(undefined))
            style={
              Object {
                "color": "black",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            timeout
          </ForwardRef(withTheme(undefined))>
          <ForwardRef(withTheme(undefined))
            style={
              Object {
                "color": "black",
                "fontSize": 16,
              }
            }
          >
            0
          </ForwardRef(withTheme(undefined))>
        </Component>
        <Component
          key="2"
          style={
            Object {
              "paddingHorizontal": 16,
              "paddingVertical": 8,
            }
          }
        >
          <ForwardRef(withTheme(undefined))
            style={
              Object {
                "color": "black",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            url
          </ForwardRef(withTheme(undefined))>
          <ForwardRef(withTheme(undefined))
            style={
              Object {
                "color": "black",
                "fontSize": 16,
              }
            }
          >
            https://run.mocky.io/v3/1a2d092a-42b2-4a89-a44f-267935dc13e9
          </ForwardRef(withTheme(undefined))>
        </Component>
        <Subheading
          style={
            Object {
              "backgroundColor": "lightgray",
              "borderWidth": 0,
              "color": "black",
              "fontWeight": "bold",
              "marginTop": -1,
              "paddingHorizontal": 16,
              "paddingVertical": 8,
            }
          }
        >
          REQUEST HEADERS
        </Subheading>
        <Subheading
          style={
            Object {
              "backgroundColor": "lightgray",
              "borderWidth": 0,
              "color": "black",
              "fontWeight": "bold",
              "marginTop": -1,
              "paddingHorizontal": 16,
              "paddingVertical": 8,
            }
          }
        >
          REQUEST DATA
        </Subheading>
        <Component
          style={
            Object {
              "paddingHorizontal": 16,
              "paddingVertical": 8,
            }
          }
        >
          <ForwardRef(withTheme(undefined))
            style={
              Object {
                "color": "black",
                "fontSize": 16,
              }
            }
          >
            dataSent
          </ForwardRef(withTheme(undefined))>
        </Component>
        <Subheading
          style={
            Object {
              "backgroundColor": "lightgray",
              "borderWidth": 0,
              "color": "black",
              "fontWeight": "bold",
              "marginTop": -1,
              "paddingHorizontal": 16,
              "paddingVertical": 8,
            }
          }
        >
          RESPONSE HEADERS
        </Subheading>
        <Subheading
          style={
            Object {
              "backgroundColor": "lightgray",
              "borderWidth": 0,
              "color": "black",
              "fontWeight": "bold",
              "marginTop": -1,
              "paddingHorizontal": 16,
              "paddingVertical": 8,
            }
          }
        >
          RESPONSE BODY
        </Subheading>
        <Component
          style={
            Object {
              "paddingHorizontal": 16,
              "paddingVertical": 8,
            }
          }
        >
          <ForwardRef(withTheme(undefined))
            style={
              Object {
                "color": "black",
                "fontSize": 16,
              }
            }
          >
            response
          </ForwardRef(withTheme(undefined))>
        </Component>
      </Component>
    </Component>
  </Component>
</Component>
`;
